writing/2019-10-05-Haskell-in-Production.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Haskell in Production
+date: 2019-10-05
+published: true
+---
+
+# Haskell in Production
+During the past year, my team has been building production services using
+Haskell. It's been quite a journey. None of us had written production code
+using Haskell before. We were all familiar with the language and excited to
+get to assess it in production. Because that's really what we've been doing;
+seeing if Haskell is a good fit for us in order to deliver quickly and with
+great quality.
+
+## tl;dr
+Yes, Haskell is great for business. Hiring is easy, the ecosystem is great, the
+libraries are mature and the code we've produced makes it easy to onboard new
+folks.
+
+## The longer version
+Knowing *exactly* how we wanted to write production services using this
+language was not as straight-forward.
+
+We tried a lot of different patterns - readers, handlers, MTL, and tagless
+final. You've surely heard of some of these, and they all have their own pros
+and cons. So which one should you choose?
+
+That is why I've written this article series, to help you get a good sense of
+how production Haskell is written and what to avoid along the road.
+
+## Existing parts in this series
+
+* Part 0: Introduction (this page!)
+* Part 1: Making components testable
+<!-- * Part 2: Testing your components -->
+<!-- * Part 3: Deploying your application -->
+
+This section will be updated as more articles are published.
+
+## Companion repo
+For this series you can find the code that goes along with each part in the
+[haskell-in-production](https://github.com/felixmulder/haskell-in-production)
+repo on GitHub.
+
+## Concerns when writing a production service
+We have a couple of concerns that we would have no-matter which language we're
+writing our service in.
+
+* Multiple environments and regions
+* Testability
+* Performance & scalability
+* Maintainability
```

Comment:
Is logging/traceability covered in one of those?
In my experience that's the first step from play to production.